apps/desktop-wallet/src/modals/AddressOptionsModal.tsx

```diff
@@ -37,19 +39,19 @@ import { saveAddressSettings } from '@/storage/addresses/addressesStorageUtils'
 import { getName } from '@/utils/addresses'
 import { getRandomLabelColor } from '@/utils/colors'
 
-interface AddressOptionsModalProps {
+export interface AddressOptionsModalProps {
   addressHash: AddressHash
-  onClose: () => void
 }
 
-const AddressOptionsModal = ({ addressHash, onClose }: AddressOptionsModalProps) => {
+const AddressOptionsModal = memo(({ id, addressHash }: ModalBaseProp & AddressOptionsModalProps) => {
```

Comment:
About the use of memoization: I've experimented with a runtime check giving a warning if one modal isn't memoized, and it seems to work :) 
I wonder what you think about it. Yes is uses the `Children` api, but that's only to output a warning, so no risk there. 

Check out [this commit](https://github.com/alephium/alephium-frontend/commit/0b08a802661a15e4bdb72e0c9a967187de1b7270) 👀

Reply:
It could also be added to `ModalContainer` actually, avoiding the use of `Children`.

Reply:
I [implemented your proposal](https://github.com/alephium/alephium-frontend/pull/815/commits/16c343820d1d6dba2c39a925ccbd68d07998bb18) in this commit and added you as a co-author! Good job! 🤝 

I tweaked it a little bit to:
- Only run when needed
- Share it through out `@alephium/shared-react` package